exporters/metrics/util/src/main/java/io/opencensus/exporter/metrics/util/QueueMetricProducer.java

```diff
@@ -0,0 +1,196 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.metrics.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.common.Scope;
+import io.opencensus.metrics.export.Metric;
+import io.opencensus.metrics.export.MetricProducer;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.Sampler;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.samplers.Samplers;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Queue;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Wrapper of {@link MetricProducer} which allows metrics to be pushed and buffered.
+ *
+ * @since 0.20
+ */
+public final class QueueMetricProducer extends MetricProducer {
+
+  private static final Object monitor = new Object();
+  private static final Tracer tracer = Tracing.getTracer();
+  private static final Sampler probabilitySampler = Samplers.probabilitySampler(0.0001);
+
+  private static final String DEFAULT_SPAN_NAME = "PushMetrics";
+  private static final int DEFAULT_BUFFER_SIZE = 32;
+  private static final String ATTRIBUTE_KEY_DROPPED_METRICS = "DroppedMetrics";
+
+  @GuardedBy("monitor")
+  private final Queue<Metric> bufferedMetrics = new LinkedList<>();
+
+  private final int bufferSize;
+  private final String spanName;
+
+  private QueueMetricProducer(int bufferSize, String spanName) {
+    this.bufferSize = bufferSize;
+    this.spanName = spanName;
+  }
+
+  /**
+   * Creates a new {@link QueueMetricProducer}.
+   *
+   * @param options the options for {@link QueueMetricProducer}.
+   * @return a {@code QueueMetricProducer}.
+   * @since 0.20
+   */
+  public static QueueMetricProducer create(Options options) {
+    checkNotNull(options, "options");
+    checkArgument(options.getBufferSize() > 0, "buffer size should be positive.");
+    return new QueueMetricProducer(options.getBufferSize(), options.getSpanName());
+  }
+
+  /**
+   * Pushes {@link Metric}s to this {@link QueueMetricProducer}.
+   *
+   * <p>When buffer of this {@link QueueMetricProducer} is full, the oldest {@link Metric}s will be
+   * dropped.
+   *
+   * @param metrics {@code Metrics} to be added to this {@code QueueMetricProducer}.
+   * @since 0.20
+   */
+  public void pushMetrics(Collection<Metric> metrics) {
+    synchronized (monitor) {
+      Span span =
```

Comment:
No reason to create a span for this small amount of work.

Reply:
This span is mainly for recording the number of dropped metrics: https://github.com/census-instrumentation/opencensus-java/pull/1804/files#diff-626f7bea3252f1a550264d2b41173952R100.

Or would you suggest just logging the number here?

Reply:
Just don't do anything for the moment. We need to define the "monitoring" story for our library. Things like what metrics every exporter should record, etc.

Reply:
Make sense, removed span creation.